weechat-rs/src/bar.rs

```diff
@@ -0,0 +1,85 @@
+use core::ptr;
+use libc::c_char;
+use std::os::raw::c_void;
+use weechat_sys::{
+    t_gui_bar_item, t_gui_buffer, t_gui_window, t_hashtable, t_weechat_plugin,
+};
+
+use crate::{Buffer, LossyCString, Weechat};
+
+struct BarItemCbData {
+    callback: fn(item: &BarItem, buffer: &Buffer) -> String,
+    weechat_ptr: *mut t_weechat_plugin,
+}
+
+pub struct BarItem {
```

Comment:
Documentation please. Mention if the item will get removed if the object gets dropped. I do think we should remove the bar item if this gets dropped. So this is missing also a drop implementation.

Otherwise people will have a nasty crash if they drop the object and the pointers get removed.